hub/policy/src/main/java/uk/gov/ida/hub/policy/domain/IdpSelected.java

```diff
@@ -23,10 +25,11 @@ public IdpSelected(String selectedIdpEntityId, String principalIpAddress) {
         this.principalIpAddress = principalIpAddress;
     }
 
-    public IdpSelected(String selectedIdpEntityId, String principalIpAddress, Boolean registration) {
+    public IdpSelected(String selectedIdpEntityId, String principalIpAddress, Boolean registration, LevelOfAssurance requestedLoa) {
```

Comment:
Hmm - this constructor is starting to take quite a few params.

I tend to find that calling long constructors in Java gets very confusing very fast (no named params 😭 - although to be fair Intellij is trying to work around this now).
The standard Java pattern is to use a builder to wrap the constructor, ie

```
new Foo(true, true, false, true);
``` 
becomes
```
aFoo()
  .withEnabled(true)
  .withImage(true)
  .withRegistration(false)
  .withParent(true)
  .build()
```
which is quite a lot more obvious (if a bit more verbose).

I don't think that this is a pattern that we've used anywhere in verify though (we only seem to use builders in our test code) and I'm not sure that this specific case is a good enough reason to add a whole new pattern.

What do you think? If you have another idea that would also be great!
(And sorry if you know all this already - let me know and I'll shut up a bit! 😄 )

Reply:
Fair point but I think it's fine in this case because 1) It's got _only_ 4 parameters (doesn't seem to be too many to me) 2) This constructor is only used in code in tests, and it's really there only for the JSON/Javax validation libraries when translating POST messages from frontend into Java objects